service/src/main/java/com/derpgroup/quip/model/QuipStore.java

```diff
@@ -0,0 +1,107 @@
+package com.derpgroup.quip.model;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.derpgroup.quip.configuration.QuipConfig;
+import com.derpgroup.quip.manager.QuipRequestTypes;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class QuipStore {
+  private final Logger LOG = LoggerFactory.getLogger(QuipStore.class);
+  
+  private static QuipStore instance;
+  private boolean initialized = false;
+  
+  Map<QuipRequestTypes, String> sourceFiles = new HashMap<QuipRequestTypes,String>();
+  Map<QuipRequestTypes, Instant> lastUpdateTimes = new HashMap<QuipRequestTypes,Instant>();
+  Map<QuipRequestTypes, List<Quip>> quips = new HashMap<QuipRequestTypes,List<Quip>>();
+  int refreshRate;  // seconds
+
+  private QuipStore(){}
+  
+  public static synchronized QuipStore getInstance(){
+    if(instance == null){
+      instance = new QuipStore();
+    }
+    return instance;
+  }
+  
+  public synchronized void init(QuipConfig config){
+    if(initialized){
+      throw new RuntimeException("QuipStore is already initialized");
+    }
+    sourceFiles.put(QuipRequestTypes.COMPLIMENT, config.getComplimentsFile());
+    sourceFiles.put(QuipRequestTypes.WINSULT, config.getWinsultsFile());
+    sourceFiles.put(QuipRequestTypes.INSULT, config.getInsultsFile());
+    sourceFiles.put(QuipRequestTypes.BACKHANDED_COMPLIMENT, config.getBackhandedComplimentsFile());
+    refreshRate = config.getRefreshRate();
+    initialized = true;
+    try{
+      updateQuips(QuipRequestTypes.COMPLIMENT);
+      updateQuips(QuipRequestTypes.WINSULT);
+      updateQuips(QuipRequestTypes.INSULT);
+      updateQuips(QuipRequestTypes.BACKHANDED_COMPLIMENT);
+      LOG.info("QuipStore initialized");
+    }
+    catch(IOException e){
+      LOG.error("There was a problem loading quips",e);
+    }
+  }
+  
+  protected void updateQuips(QuipRequestTypes quipType) throws IOException{
+    if(!initialized){throw new RuntimeException("QuipStore must be initialized before use");}
+
+    synchronized(quipType){
+      int oldQuipCount = 0;
+      if(quips.get(quipType)!=null){
+        oldQuipCount = quips.get(quipType).size();
+      }
+
+      quips.put(quipType,readQuipsFromFile(sourceFiles.get(quipType)));
```

Comment:
Should we move the try/catch to surround this section, rather than make the entire function throw it